test/unit/assorted/server_discovery_and_monitoring.spec.test.ts

```diff
@@ -232,7 +234,6 @@ async function executeSDAMTest(testData) {
         }
       }
 
-      topology.removeAllListeners('error');
```

Comment:
I actually don't think it's correct to remove this - any test that has more than one phase would get multiple handlers assigned without that error listener, so your array would get double the data, but you couldn't catch it because we only assert on the event types and the message, not on the count of the registered events. So maybe we need to add an assertion on the length of `errorEvents`

Reply:
I checked the length and moved the listener outside the loop to make sure it's not added more than once

Reply:
Two concerns:
1) The check you added doesn't guarantee that we don't get more events than we should, so it's guarding against no events being registered but not guarding against events getting doubled up, for example;

2) before, we were only attaching the handler and collecting events per phase, which makes sense if we're trying to isolate what is happening each time; *now* we have one listener and one giant pile of errorEvents the entirety of which gets checked for every phase regardless of when the particular event was caught, and there is no reset, so every phase adds more events into the pile and then checks all the existing events and the new events - that seems wrong

Reply:
I think it's fine to leave the error handler attachment at the top, but the collected errors definitely need to be cleared at the end of each phase, and I would still like to find a way to assert on the number of those expected error events for each phase